pymc3/distributions/distribution.py

```diff
@@ -309,7 +309,7 @@ def draw_values(params, point=None, size=None):
                 # param was drawn in related contexts
                 v = drawn[(p, size)]
                 evaluated[i] = v
-            elif name is not None and name in point:
+            elif name is not None and hasattr(p, 'model') and name in point:
```

Comment:
i don't see where `p.model` is used!

Reply:
It was really strange for me too, but it is the only thing truly needed for the fix. I copied the way in which `_draw_value` decides whether to grab the value from the `point` dict or not. Basically, it works because only `*RV` instances have a model attribute, while the deterministics are just theano variables with no model attribute. This means that the deterministics would never be picked from `point` even though their name-value pair is in the dictionary.

Reply:
can you add a comment above explaining that? something like the below would suffice:

```
# filter out Deterministics by checking for `model` attribute
```